ta4j-core/src/main/java/org/ta4j/core/BaseTradingRecord.java

```diff
@@ -226,13 +228,31 @@ public boolean exit(int index, Num price, Num amount) {
 
     @Override
     public List<Position> getPositions() {
-        return positions;
+        return new LinkedList<>(sortedPositions.values());
+    }
+
+    @Override
+    public Position getLastPositionRegardingIndex(int index) {
```

Comment:
rename to 

`public Position getLastPosition(int index){..}`

The "regarding index" is already implicitly told by its needing parameter list `(int index)`